### DISCUSSION TITLE
Cura engine wrapper

### DISCUSSION THREAD

**Discussion by agusventuri** [ErwinRieger/ddprint#7] [Q&A] [answered]
Hey! I'm trying to develop an automatic budgeter using CuraEngine but I'm lost on how to use the generated profiles. That's when I stumbled upon your wrapper and that's exactly what I need. The problem is... I can't get it to run, even using Cura 2.3.x.
Can you point me in the right direction?
What I did first was:
1. Download Cura (v2.3.1) ([source code](https://github.com/Ultimaker/Cura/releases/tag/2.3.1)
2. Execute `python3 cura-engine-wrapper.py ~/Cura-2.3.1/cura/`

But I was getting errors regarding `UM.Application` and other imports not working. Then I did:
1. Download Uranium (v2.3.1) ([source code](https://github.com/Ultimaker/Uranium/releases/tag/2.3.1)
2. Move `cura-engine-wrapper.py` inside Uranium folder
3. Execute `python3 cura-engine-wrapper.py ~/Cura-2.3.1/cura/`

Now the `UM.x` errors are gone as it's correctly importing that stuff now but now it fails when importing `cura.Settings`. What I did was move everything from the downloaded Cura folder to the Uranium folder. This did the trick but it now fails as it can't find plugin XmlMaterialProfile. I moved every plugin from Cura to Uranium's plugin folder but it still fails :(
I don't know what I'm doing wrong and I also don't exactly know how Cura and Uranium's framework works. 
If you can point me in the right direction I'll be really thankful. If I can update the wrapper to a newer cura version I'll create a Pull request for you to review.
Thanks

**Comment by ErwinRieger** [comment, marked as answer]
Hi, no need to copy things around...

You have to set/export the PYTHONPATH and LD_LIBRARY_PATH variables in your einvironment, see cura-engine-wrapper.sh.

Note: i'm not using this wrapper any more, so i can't help with newer versions of cura engine...

See also: https://ibrieger.de/cura-commandline-wrapper.html

**Reply by agusventuri** [reply]
Thank you for your answer! I'll try to do it this way on a fresh install of Cura 2.3
